hudi-common/src/main/java/org/apache/hudi/common/util/SpillableMapUtils.java

```diff
@@ -154,8 +155,13 @@ private static Object getPreCombineVal(GenericRecord rec, String preCombineField
     if (preCombineField == null) {
       return 0;
     }
-    Schema.Field field = rec.getSchema().getField(preCombineField);
-    return field == null ? 0 : rec.get(field.pos());
+    // Handle multiple ordering/preCombineFields
+    if (preCombineField.contains(";")) {
+      return HoodieAvroUtils.getMultipleNestedFieldVals(rec, preCombineField, false);
+    } else {
```

Comment:
Let's move the support for multiple precombine fields into another JIRA/PR, let's focus on partial update in this one.

Reply:
OK,  I revert all changes wrt multiple ordering support,  will create another PR since I feel we still need to support out-of-order events scenario

Reply:
for the whole feature can refer to https://github.com/apache/hudi/compare/master...fengjian428:hudi:partial-update-feature